framework/src/auxkernels/ProjectionAux.C

```diff
@@ -0,0 +1,113 @@
+  params.set<bool>("_allow_nodal_to_elemental_coupling") = true;
+
+  // We need some ghosting for all elemental to nodal projections
+  params.addParam<unsigned short>("ghost_layers", 1, "The number of layers of elements to ghost.");
+  params.addRelationshipManager(
+      "ElementPointNeighborLayers",
+      Moose::RelationshipManagerType::GEOMETRIC | Moose::RelationshipManagerType::ALGEBRAIC,
+      [](const InputParameters & obj_params, InputParameters & rm_params)
+      {
+        rm_params.set<unsigned short>("layers") = obj_params.get<unsigned short>("ghost_layers");
+        rm_params.set<bool>("use_displaced_mesh") = obj_params.get<bool>("use_displaced_mesh");
+      });
+  return params;
+}
+
+ProjectionAux::ProjectionAux(const InputParameters & parameters)
+  : AuxKernel(parameters),
+    _v(coupledValue("v")),
+    _source_variable(getVar("v", 0)),
+    _source_sys(_source_variable->sys())
+{
+  // Output some messages to user
+  if (isNodal() && _source_variable->isNodal() && _source_variable->order() < _var.order())
+    mooseInfo(
+        "Target nodal variable is higher order than source, using volume-weighted projection");
+  else if (_source_variable->order() > _var.order())
+    mooseInfo("Projection lowers order, please expect a loss of accuracy");
+  else if (_var.feType().family == SIDE_HIERARCHIC)
+    paramError("variable", "SIDE_HIERARCHIC is not supported");
+}
+
+Real
+ProjectionAux::computeValue()
+{
+  if (!isNodal() || (_source_variable->isNodal() && _source_variable->order() >= _var.order()))
+    return _v[_qp];
+  // projecting continuous elemental variable onto a node
+  else if (!_source_variable->isNodal() && _source_variable->getContinuity() != DISCONTINUOUS &&
+           _source_variable->getContinuity() != SIDE_DISCONTINUOUS)
+    return _source_sys.system().point_value(
+        _source_variable->number(), *_current_node, elemOnNodeVariableIsDefinedOn());
+  // Handle discontinuous elemental variable projection into a nodal variable
+  // AND nodal low order -> nodal higher order
+  else
+  {
+    // Custom projection rule : use neighbor element centroid values weighted by element volumes
+    // First, find all the elements that this node is part of
+    auto elem_ids = _mesh.nodeToElemMap().find(_current_node->id());
+    mooseAssert(elem_ids != _mesh.nodeToElemMap().end(),
+                "Should have found an element around node " + std::to_string(_current_node->id()));
+
+    // Get the neighbor element centroid values & element volumes
+    std::vector<Real> elem_values(elem_ids->second.size(), 0);
+    std::vector<Real> elem_volumes(elem_ids->second.size(), 0);
+    auto index = 0;
+    for (auto & id : elem_ids->second)
+    {
+      const auto & elem = _mesh.elemPtr(id);
+      if (_source_variable->hasBlocks(elem->subdomain_id()))
+      {
+        elem_values[index] = _source_sys.system().point_value(
+            _source_variable->number(), elem->true_centroid(), elem);
```

Comment:
Umm you're right.
I added the second option last, and did not think to move that case there.

In my mind it was fair to weight the centroid value with the volume. But for using the (discontinuous) nodal value, I would have to weight with either something like the local solid angle of the element or some distance to the centroid instead of the volume
I changed it. I just want a projection rule, it will have to be approximate anyway